pkg/operator/staticpod/certsyncpod/certsync_controller.go

```diff
@@ -197,12 +197,19 @@ func (c *CertSyncController) sync() error {
 				continue
 			}
 
+			// check if the secret file exists, skip firing events if it does not
+			secretFile := getSecretDir(c.destinationDir, s.Name)
+			if _, err := os.Stat(secretFile); os.IsNotExist(err) {
+				continue
+			}
+
 			// remove missing content
-			if err := os.RemoveAll(getSecretDir(c.destinationDir, s.Name)); err != nil {
-				c.eventRecorder.Warningf("CertificateUpdateFailed", "Failed removing file for secret: %s/%s: %v", secret.Namespace, secret.Name, err)
+			if err := os.RemoveAll(secretFile); err != nil {
+				c.eventRecorder.Warningf("CertificateUpdateFailed", "Failed removing file for missing secret: %s/%s: %v", secret.Namespace, secret.Name, err)
 				errors = append(errors, err)
+				continue
 			}
-			c.eventRecorder.Warningf("CertificateRemoved", "Removed file for secret: %s/%s", secret.Namespace, secret.Name, err)
+			c.eventRecorder.Warningf("CertificateRemoved", "Removed file for missing secret: %s/%s", secret.Namespace, secret.Name)
 			continue
```

Comment:
nit: I think that we don't need `continue` at the end of `case` clause.

Reply:
@p0lyn0mial i think we do, otherwise it will escape down :-)

Reply:
ah, ic, we would escape `switch` statement :)